diff -Naur xbmc-147cec4077415c93d3d84fb82cb6b695b5a9094c/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp xbmc/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
--- xbmc-147cec4077415c93d3d84fb82cb6b695b5a9094c/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2017-05-24 23:49:32.000000000 +0300
+++ xbmc/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2017-06-14 21:21:13.000000000 +0300
@@ -38,6 +38,7 @@
 #include "settings/AdvancedSettings.h"
 #if defined(HAS_LIBAMCODEC)
 #include "utils/AMLUtils.h"
+#include "utils/SysfsUtils.h"
 #endif
 
 #ifdef TARGET_POSIX
@@ -538,6 +539,41 @@
   if (aml_present())
   {
     aml_set_audio_passthrough(m_passthrough);
+
+
+    int aml_digital_codec;
+    switch(format.m_streamInfo.m_type)
+    {
+      case CAEStreamInfo::STREAM_TYPE_AC3:
+        aml_digital_codec = 2;
+        break;
+
+      case CAEStreamInfo::STREAM_TYPE_DTS_512:
+      case CAEStreamInfo::STREAM_TYPE_DTS_1024:
+      case CAEStreamInfo::STREAM_TYPE_DTS_2048:
+      case CAEStreamInfo::STREAM_TYPE_DTSHD_CORE:
+        aml_digital_codec = 3;
+        break;
+
+      case CAEStreamInfo::STREAM_TYPE_EAC3:
+        aml_digital_codec = 4;
+        break;
+
+      case CAEStreamInfo::STREAM_TYPE_DTSHD:
+        aml_digital_codec = 8;
+        break;
+
+      case CAEStreamInfo::STREAM_TYPE_TRUEHD:
+        aml_digital_codec = 7;
+        break;
+
+      default:
+        if (inconfig.channels > 2)
+          aml_digital_codec = 6;
+        else
+          aml_digital_codec = 0;
+    }
+    SysfsUtils::SetInt("/sys/class/audiodsp/digital_codec", aml_digital_codec);
   }
 #endif
 
